src/pytorch_lightning/_graveyard/profiler.py

```diff
@@ -0,0 +1,36 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytorch_lightning.profiler.base as base
+
+
+class AbstractProfiler:
+    # TODO: Remove in v2.0.0
+    def __init__(self, *_, **__):
+        raise NotImplementedError(
```

Comment:
well, this exception mean the same as being removed, so nit sure what is this about :/

Reply:
Not exactly the same, IMO. We've recently updated our deprecation policy slightly in #15049 so that users would keep seeing this kind of helpful exceptions until 2.0 instead of seeing random errors caused by API removals.

Reply:
LGTM!